src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASubmissionStatus.tsx

```diff
@@ -0,0 +1,315 @@
+    </Box>
+  )
+}
+
+const submissionStatusFragment = graphql`
+  fragment MyCollectionArtworkSWASubmissionStatus_artwork on Artwork {
+    listedArtworksConnection(first: 1) {
+      edges {
+        node {
+          internalID
+        }
+      }
+    }
+    consignmentSubmission {
+      internalID
+      state
+      stateLabel
+      stateHelpMessage
+    }
+  }
+`
+
+interface SubmissionStatusModalProps {
+  show: boolean
+  onClose(): void
+  submissionData: SubmissionData
+}
+
+const SubmissionStatusModal: React.FC<SubmissionStatusModalProps> = ({
+  show,
+  onClose,
+  submissionData,
+}) => {
+  if (!show) return null
+
+  return (
+    <ModalDialog onClose={onClose}>
+      <Box>
+        <Text variant="xs" mb={1}>
+          Submission Status
+        </Text>
+
+        <Text variant="md" fontWeight="400">
+          {submissionData.stateLabel}
+        </Text>
+
+        {!!submissionData.actionLabel && (
+          <Text variant="md" fontWeight="400" color="orange100">
+            {submissionData.actionLabel}
+          </Text>
+        )}
+
+        <Spacer y={1} />
+
+        <Text variant="sm" color="black60">
+          {submissionData.description}
+        </Text>
+
+        <Spacer y={2} />
+
+        {!!submissionData.button && (
+          <Button
+            variant={submissionData.button.variant}
+            // @ts-ignore
+            as={RouterLink}
+            onClick={() => {
+              // TODO: Tracking
+            }}
+            to={submissionData.button.to}
+            width="100%"
+            mb={2}
+          >
+            {submissionData.button.label}
+          </Button>
+        )}
+      </Box>
+    </ModalDialog>
+  )
+}
+
+// TODO: Get the description and state label from Metaphysics once it is updated
+const getSubmissionData = (
+  artwork: MyCollectionArtworkSWASubmissionStatus_artwork$data
+): SubmissionData | null => {
+  console.log({ artwork })
+  const submission = artwork.consignmentSubmission
+
+  if (!submission) return { stateLabel: null }
+
+  const listedArtwork = extractNodes(artwork.listedArtworksConnection)[0]
+
+  // The artwork has been listed on Artsy.
+  if (listedArtwork) {
+    return {
+      button: listedArtwork
+        ? {
+            label: "View Listing",
+            to: `/artwork/${listedArtwork?.internalID}`,
+            variant: "secondaryBlack",
+          }
+        : undefined,
+      description: "Your artwork has been successfully listed on Artsy.",
+      stateLabel: "Listed",
+    }
+  }
+
+  switch (submission.state) {
+    case "DRAFT":
+      return {
+        button: {
+          label: "Complete Submission",
+          to: `/sell/submissions/${submission.internalID}/${INITIAL_STEP}`,
+          variant: "primaryBlack",
+        },
+        description:
+          "You’ve started a submission to sell with Artsy but have not yet completed it.",
+        stateLabel: null,
+        actionLabel: "Complete Submission",
+      }
+    case "SUBMITTED":
```

Comment:
What do you think about adding this to MP? I see we already have some logic in ArtworkConsignmentSubmissionType

Reply:
Yes, good catch! We should definitely update and use the copy from MP. There should already be a ticket for it: https://artsy.slack.com/archives/C05EQL4R5N0/p1721746558213349?thread_ts=1721746117.454679&cid=C05EQL4R5N0

Reply:
Perfect 👍

Reply:
done 👍